enhancements/cluster-logging/forward_to_cloudwatch.md

```diff
@@ -0,0 +1,249 @@
+
+For *node-scoped* logs we will create "audit"" and "infrastructure" streams for each node: e.g. `node-id/audit`.
+
+By default the forwarder creates 3 *log groups* for each of the log categories:  "application", "infrastructure" and "audit".
+Application logs can also be grouped by namespace, k8s labels, or forwarder configuration, see below.
+
+### API fields
+
+New API fields in the `output.cloudwatch` section:
+
+- `region`: (string) AWS region name, required to connect.
+- `retentionDays`: (number) Number of days to keep logs. Defaults to cloudwatch default retention.
+- `groupByNamespace`: (bool) If true, set the Cloudwatch group name to the source namespace.
+  If false, group name is set to the log category: "application", "infrastructure" or "audit".
+
+*Nice-to-have*: the following fields may be omitted from the first release and implemented later.
+
+- `group`: (string, *nice-to-have*) Fixed Cloudwatch log group name for all logs leaving this this output.
+- `groupByLabel`: (string, *nice-to-have*) If this Kubernetes label key is present, use the value as group name.
+  If the label is absent, set the group name as if the `groupByLabel` field was absent.
+
+Existing fields:
+
+- `url`: Not used in production. Sets the `endpoint` parameter in fluentd, which can apparently be used to connect to a local test server.
+- `secret`: AWS credentials, keys `aws_access_key_id` and `aws_secret_access_key`.
+
+**Note**: The installer UI (Addon or OLM) can get AWS credentials from a `cloudcredential.openshift.io/v1`.
+The user only has to provide a `region` to enable cloudwatch forwarding for a cluster.
+Details are out of scope for this proposal.
+
+### User Stories
+
+#### I want to forward logs to Cloudwatch instead of a local store
+
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+spec:
+  outputs:
+   - name: CloudwatchOut
+     type: cloudwatch
+     cloudwatch:
+       region: myregion
+     secret:
+        name: mysecret
+  pipelines:
+  - inputRefs: [application, infrastructure, audit]
+  - outputRefs: [CloudwatchOut]
+```
+
+#### I want to group application logs by namespace
+
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+spec:
+  outputs:
+   - name: CloudwatchOut
+     type: cloudwatch
+     cloudwatch:
+       region: myregion
+	   groupByNamespace: true
+     secret:
+        name: mysecret
+  pipelines:
+  - inputRefs: [application, infrastructure, audit]
+  - outputRefs: [CloudwatchOut]
+```
+
+#### I want to define my own log groups (nice-to-have)
+
+There are two ways to define custom log groups:
+- Use a Kubernetes label with the group name.
+- Define the group name in the `ClusterLogForwarder.Output`
+
+1. Using a kubernetes label named `LogGroup`
+
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+spec:
+  outputs:
+   - name: CloudwatchOut
+     type: cloudwatch
+     cloudwatch:
+       region: myregion
+	   groupByLabel: LogGroup
```

Comment:
What happens if the user specifies `groupByLabel` and `groupByNamespace`?

Reply:
Illegal, I'll document it.